python/cudf_polars/cudf_polars/dsl/expressions/boolean.py

```diff
@@ -31,22 +30,49 @@
 __all__ = ["BooleanFunction"]
 
 
+class BooleanFunctionName(Enum):
+    All = auto()
+    AllHorizontal = auto()
+    Any = auto()
+    AnyHorizontal = auto()
+    IsBetween = auto()
+    IsDuplicated = auto()
+    IsFinite = auto()
+    IsFirstDistinct = auto()
+    IsIn = auto()
+    IsInfinite = auto()
+    IsLastDistinct = auto()
+    IsNan = auto()
+    IsNotNan = auto()
+    IsNotNull = auto()
+    IsNull = auto()
+    IsUnique = auto()
+    Not = auto()
+
+    @staticmethod
+    def get_polars_type(tp: BooleanFunctionName):
+        function, name = str(tp).split(".")
+        if function != "BooleanFunction":
+            raise ValueError("BooleanFunction required")
+        return getattr(BooleanFunctionName, name)
+
+
 class BooleanFunction(Expr):
     __slots__ = ("name", "options")
     _non_child = ("dtype", "name", "options")
 
     def __init__(
         self,
         dtype: plc.DataType,
-        name: pl_expr.BooleanFunction,
+        name: BooleanFunctionName,
         options: tuple[Any, ...],
         *children: Expr,
     ) -> None:
         self.dtype = dtype
         self.options = options
         self.name = name
         self.children = children
-        if self.name == pl_expr.BooleanFunction.IsIn and not all(
+        if self.name == BooleanFunctionName.IsIn and not all(
```

Comment:
*nit*: Because we have enums, we should now use `is` for comparison.

Reply:
Done in https://github.com/rapidsai/cudf/pull/17418/commits/7d009f9578f554c51ddbac0b6120496956339042 .